pkg/lang/javascript/aws_runtime/_fs.ts

```diff
@@ -0,0 +1,163 @@
+const path = require('path')
+import { Readable } from 'stream'
+import {
+    S3Client,
+    PutObjectCommand,
+    GetObjectCommand,
+    ListObjectsCommand,
+    ListObjectsCommandInput,
+    DeleteObjectCommand,
+    HeadObjectCommand,
+} from '@aws-sdk/client-s3'
+
+const payloadBucketPhysicalName = process.env.KLOTHO_S3_PREFIX + '{{.PayloadsBucketName}}'
+const targetRegion = process.env['AWS_TARGET_REGION']
+
+const userBucketPath = '/files'
+
+const s3Client = new S3Client({ region: targetRegion })
+
+const streamToString = (stream: Readable): Promise<string> =>
+    new Promise((resolve, reject) => {
+        const chunks: Uint8Array[] = []
+        stream.on('data', (chunk) => chunks.push(chunk))
+        stream.on('error', reject)
+        stream.on('end', () => resolve(Buffer.concat(chunks).toString('utf8')))
+    })
+
+async function getCallParameters(paramKey, dispatcherMode) {
+    let isEmitter = dispatcherMode === 'emitter' ? true : false
+    try {
+        const bucketParams = {
+            Bucket: payloadBucketPhysicalName,
+            Key: paramKey,
+        }
+        const result = await s3Client.send(new GetObjectCommand(bucketParams))
+
+        let parameters = ''
+        if (result.Body) {
+            parameters = await streamToString(result.Body as Readable)
+        }
+
+        if (isEmitter && Array.isArray(parameters)) {
+            // Emitters only have 1 parameter - the runtime saves an array, so we
+            // normalize the parameter
+            parameters = _.get(parameters, '[0]')
+            if (Array.isArray(parameters)) {
+                let paramPairs = _.toPairs(parameters)
+                paramPairs = paramPairs.map((x) => {
+                    if (_.get(x, '[1].type') == 'Buffer') {
+                        return [x[0], Buffer.from(_.get(x, '[1].data'))]
+                    } else {
+                        return x
+                    }
+                })
+
+                parameters = _.fromPairs(paramPairs)
+            }
+        }
+
+        return parameters || {}
+    } catch (e) {
+        console.error(e)
+        return
+    }
+}
+exports.getCallParameters = getCallParameters
+
+async function saveParametersToS3(paramsS3Key, params) {
+    try {
+        const bucketParams = {
+            Bucket: payloadBucketPhysicalName,
+            Key: paramsS3Key,
+            Body: flatted.stringify(params),
+        }
+        await s3Client.send(new PutObjectCommand(bucketParams))
+    } catch (e) {
+        console.error(e)
+        return
+    }
+}
+exports.saveParametersToS3 = saveParametersToS3
+
+async function s3_writeFile(...args) {
+    const bucketParams = {
+        Bucket: payloadBucketPhysicalName,
+        Key: `${userBucketPath}/${args[0]}`,
+        Body: args[1],
+    }
+    try {
+        await s3Client.send(new PutObjectCommand(bucketParams))
+        console.log('Successfully uploaded object: ' + bucketParams.Bucket + '/' + bucketParams.Key)
```

Comment:
I wonder if this should be debug? I'd be worried about spamming the service logs with what is essentially an implementation detail (they should be able to assume that if the s3 didn't log a failure, that it succeeded).

Similar for other log lines, other than the ones within catches.

Reply:
hm, yeah i can add them all as debug